src/utils/replaceEnvVars.ts

```diff
@@ -0,0 +1,28 @@
+// TODO - Update this with the dotenv interface.
+import { DotenvConfig } from "../interfaces.ts";
+
+/**
+ * This is added here in anticipation of https://github.com/BentoumiTech/denox/issues/12 being completed.
+ * There are many time we need to inject env variables into a script.
+ * These comments are merely to give you an insight to my thinking and would be removed after a successful review.
+ */
+
+function replaceEnvVars(cmd: string | undefined, env: DotenvConfig): string {
+  const envKeys: string[] = Object.keys(env);
+
+  if (!cmd) {
+    throw new ReferenceError("A command is required by this function.");
+  }
+
+  // I'm not sure if you have a loop preference? This is the method I have become accustom to in terms of readability and performance.
+  envKeys.forEach((envKey: string) => {
+    // This is a little ugly but I like explicit variable declaration for readability.
+    const envValue: string = env[envKey];
+    // String are immutable so it doesn't matter if we directly work on the cmd string.
+    cmd = cmd?.replace(`\${${envKey}}`, envValue);
```

Comment:
What do you think about changing the format from `${ENV_NAME}` to `$ENV_NAME`?

I prefer the second as it's the way POSIX commands utilize environment variables, it's the same way you would write in a bash terminal for example.

Reply:
I am completely fine with changing that

Reply:
FYI, in POSIX shells `${ENV_NAME}` == `$ENV_NAME`.

While `$ENV_NAME` is commonly used, `${ENV_NAME}` is less ambiguous when embedded within other text (eg, "${ENV_NAME}S" vs "$ENV_NAMES"). Given the possible ambiguities, you'll want to at least allow the `${ENV_NAME}` format.